obs-vscode-extension/server/src/server.ts

```diff
@@ -0,0 +1,80 @@
+import { createConnection, ProposedFeatures, TextDocuments, InitializeParams, TextDocumentSyncKind, TextDocumentChangeEvent, Position, Diagnostic, Range, DiagnosticSeverity } from "vscode-languageserver";
+import { TextDocument } from "vscode-languageserver-textdocument";
+
+import * as process from "child_process";
+import { promises as fs } from "fs";
+
+const connection = createConnection(ProposedFeatures.all);
+const documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument);
+
+connection.onInitialize((params: InitializeParams) => {
+  return {
+    capabilities: {
+      textDocumentSync: TextDocumentSyncKind.Full,
+      // completionProvider: {
+      //   triggerCharacters: ["."],
+      //   resolveProvider: false
+      // }
+    }
+  }
+});
+
+async function validateTextDocument(change: TextDocumentChangeEvent<TextDocument>) {
+  connection.console.log("Validating document");
+
+  // Create temporary file containing the current set of changes 
+  // TODO: Modify obsidianc to accept source code via stdin instead 
+  // Note: obsidianc crashes if the source file begins with a dot
+  const tmpName = __dirname + "/ObsidianLSPTmp.obs"
```

Comment:
Sounds like this might be better: https://www.npmjs.com/package/tmp

Reply:
This turns out to only be a problem if somehow the user is editing two files in parallel. Since that scenario is basically impossible, adding a new dependency seemed overkill